stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -387,6 +390,12 @@ function nnz(lu::UmfpackLU)
     return Int(lnz + unz)
 end
 
+function LinearAlgebra.issuccess(lu::UmfpackLU)
+    U, status = umfpack_numeric!(lu, false)
+    return status == UMFPACK_OK
+end
```

Comment:
I haven't worked through the `umfpack_numeric!` logic, but presume that calling this `issuccess` method on a completed `UmfpackLU` will not result in an additional numerical decomposition phase?

Reply:
I was wondering the same when I wrote this, my tests suggested it was an quick return, but I tried again now (with larger matrices) and that does not seem to be the case... Might have to store `status` in the struct after all!

Reply:
Browsed through the source and indeed it doesn't look like it is returning early if the factorization is already computed so storing the `status`. UMFPACK stores an estimate of the reciprocal condition number and this number is used to determine singularity but I don't think we have a way to extract that estimate.

Reply:
Done; I think the diff is smaller for this implementation as well.